src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -39,385 +52,213 @@
 
     private boolean includeUpper = true;
 
-    private float boost = -1;
+    private float boost = 1.0f;
 
     private String queryName;
 
+    private String format;
+
     /**
      * A Query that matches documents within an range of terms.
      *
-     * @param name The field name
-     */
-    public RangeQueryBuilder(String name) {
-        this.name = name;
-    }
-
-    /**
-     * The from part of the range query. Null indicates unbounded.
-     */
-    public RangeQueryBuilder from(Object from) {
-        this.from = from;
-        return this;
-    }
-
-    /**
-     * The from part of the range query. Null indicates unbounded.
+     * @param fieldname The field name
      */
-    public RangeQueryBuilder from(String from) {
-        this.from = from;
-        return this;
+    public RangeQueryBuilder(String fieldname) {
+        this.fieldname = fieldname;
     }
 
-    /**
-     * The from part of the range query. Null indicates unbounded.
-     */
-    public RangeQueryBuilder from(int from) {
-        this.from = from;
-        return this;
+    public RangeQueryBuilder() {
+        // for serialization
     }
 
     /**
-     * The from part of the range query. Null indicates unbounded.
+     * Get the field name for this query.
      */
-    public RangeQueryBuilder from(long from) {
-        this.from = from;
-        return this;
+    public String fieldname() {
+        return this.fieldname;
     }
 
     /**
      * The from part of the range query. Null indicates unbounded.
      */
-    public RangeQueryBuilder from(float from) {
-        this.from = from;
+    public RangeQueryBuilder from(Object from, boolean includeLower) {
+        this.from = convertToBytesRefIfString(from);
+        this.includeLower = includeLower;
         return this;
     }
 
     /**
      * The from part of the range query. Null indicates unbounded.
      */
-    public RangeQueryBuilder from(double from) {
-        this.from = from;
-        return this;
+    public RangeQueryBuilder from(Object from) {
+        return from(from, this.includeLower);
     }
 
     /**
-     * The from part of the range query. Null indicates unbounded.
+     * Gets the lower range value for this query.
      */
-    public RangeQueryBuilder gt(String from) {
-        this.from = from;
-        this.includeLower = false;
-        return this;
+    public Object from() {
+        return convertToBytesRefIfString(this.from);
```

Comment:
ok let's decide for one way and do the same across the board. What do we do with the term query now? I think it makes sense to return the string , let's just make sure we don't use getters ourselves and don't end up doing double conversions (I don't think we do, but let's double check) :)